starlark/src/stdlib/macros/mod.rs

```diff
@@ -93,76 +143,84 @@ macro_rules! starlark_signature_extraction {
+        starlark_signature_extraction!($args $call_stack $env * $t $(: $pt)?);
         starlark_signature_extraction!($args $call_stack $env $($rest)*);
     };
-    ($args:ident $call_stack:ident $env:ident ** $t:ident, $($rest:tt)* ) => {
-        starlark_signature_extraction!($args $call_stack $env ** $t);
+    ($args:ident $call_stack:ident $env:ident ** $t:ident $(: $pt:ty)?, $($rest:tt)* ) => {
+        starlark_signature_extraction!($args $call_stack $env ** $t $(: $pt)?);
         starlark_signature_extraction!($args $call_stack $env $($rest)*);
     };
-    ($args:ident $call_stack:ident $env:ident # $t:ident, $($rest:tt)* ) => {
-        starlark_signature_extraction!($args $call_stack $env # $t);
+    ($args:ident $call_stack:ident $env:ident # $t:ident $(: $pt:ty)?, $($rest:tt)* ) => {
+        starlark_signature_extraction!($args $call_stack $env # $t $(: $pt)?);
         starlark_signature_extraction!($args $call_stack $env $($rest)*);
     };
-    ($args:ident $call_stack:ident $env:ident ? $t:ident, $($rest:tt)* ) => {
-        starlark_signature_extraction!($args $call_stack $env ? $t);
+    ($args:ident $call_stack:ident $env:ident ? $t:ident $(: $pt:ty)?, $($rest:tt)* ) => {
+        starlark_signature_extraction!($args $call_stack $env ? $t $(: $pt)?);
         starlark_signature_extraction!($args $call_stack $env $($rest)*);
     };
-    ($args:ident $call_stack:ident $env:ident ? # $t:ident, $($rest:tt)* ) => {
-        starlark_signature_extraction!($args $call_stack $env ? # $t);
+    ($args:ident $call_stack:ident $env:ident ? # $t:ident $(: $pt:ty)?, $($rest:tt)* ) => {
+        starlark_signature_extraction!($args $call_stack $env ? # $t $(: $pt)?);
         starlark_signature_extraction!($args $call_stack $env $($rest)*);
     };
-    ($args:ident $call_stack:ident $env:ident $t:ident, $($rest:tt)* ) => {
-        starlark_signature_extraction!($args $call_stack $env $t);
+    ($args:ident $call_stack:ident $env:ident $t:ident $(: $pt:ty)?, $($rest:tt)* ) => {
+        starlark_signature_extraction!($args $call_stack $env $t $(: $pt)?);
         starlark_signature_extraction!($args $call_stack $env $($rest)*);
     };
-    ($args:ident $call_stack:ident $env:ident # $t:ident = $e:expr, $($rest:tt)* ) => {
-        starlark_signature_extraction!($args $call_stack $env # $t = $e);
+    ($args:ident $call_stack:ident $env:ident # $t:ident $(: $pt:ty)? = $e:expr, $($rest:tt)* ) => {
+        starlark_signature_extraction!($args $call_stack $env # $t $(: $pt)? = $e);
         starlark_signature_extraction!($args $call_stack $env $($rest)*);
     };
-    ($args:ident $call_stack:ident $env:ident $t:ident = $e:expr, $($rest:tt)* ) => {
-        starlark_signature_extraction!($args $call_stack $env $t = $e);
+    ($args:ident $call_stack:ident $env:ident $t:ident $(: $pt:ty)? = $e:expr, $($rest:tt)* ) => {
+        starlark_signature_extraction!($args $call_stack $env $t $(: $pt)? = $e);
```

Comment:
In this file below there is a documentation on how to use those macro. Maybe update it a bit with the new usages?